docs/proposals/PROPOSAL_A.md

```diff
@@ -0,0 +1,203 @@
+| allow.chflags  | freebsd.jail.allow.chflags |
+| allow.mount    | freebsd.jail.allow.mount |
+| allow.quotas    | freebsd.jail.allow.quotas |
+| allow.read_msgbuf | freebsd.jail.allow.readMsgbuf |
+| allow.socket_af | freebsd.jail.allow.socketAf |
+| allow.mlock    | freebsd.jail.allow.mlock |
+| allow.nfsd     | freebsd.jail.allow.nfsd |
+| allow.reserved_ports | freebsd.jail.allow.reservedPorts |
+| allow.unprivileged_proc_debug | - |
+| allow.suser    | freebsd.jail.allow.suser |
+| allow.mount.*  | see below            |
+| securelevel    | -                    |
+
+The jail name is set to the create command's `container-id` argument.
+
+The `devfs_ruleset` parameter is only required for jails which create new `devfs` mounts - typically OCI runtimes will mount `devfs` on the host.
+
+The `children.max` parameter is managed by the OCI runtime e.g when a new container shares namespaces with an existing container.
+
+The `allow.mount.*` parameter set is extensible - this proposal suggests representing allowed mount types as an array. As with `devfs`, typically the OCI runtime will manage mounts for the container by performing mount operations on the host.
+
+### Example
+
+An example config for a container with its own host and network namespaces. This
+container is allowed to see its own mounts and can use raw
+sockets. In addition to the minimal set of devices in the container devfs,
+`/dev/pf` is exposed, allowing the container to manage firewall rules etc. in its
+network namespace.
+
+```json
+{
+	"ociVersion": "1.1.0",
+	"hostname": "mycontainer",
+	"process": {
+		"cwd": "/",
+		"env": ["PATH=/bin:/sbin:/usr/bin:/usr/sbin"],
+		"args": ["freebsd-version"]
+	}
+	"mounts": [
+		{
+			"destination": "/dev",
+			"options": ["ruleset=4"],
+			"source": "devfs",
+			"type": "devfs"
+		},
+		{
+			"destination": "/dev/fd",
+			"source": "fdesc",
+			"type": "fdescfs"
+		}
+	],
+	"root": {
+		"path": "/path/to/container/root"
+	},
+	"freebsd": {
+		"devices": [
+			{
+				"path": "pf",
+				"mode": "0700",
+				"unhide": true
+			}
+		],
+		"jail": {
+			"host": "new",
+			"vnet": "new",
+			"enforceStatfs": 1,
+			"allow": {
+				"rawSockets": true,
+				"chflags": true
+			}
+		}
+	}
+}
+```
+
+This example shows a config for a container which is allowed to mount new tmpfs
+instances:
+
+```json
+{
+	"ociVersion": "1.1.0",
+	"hostname": "mycontainer",
+	"process": {
+		"cwd": "/",
+		"env": ["PATH=/bin:/sbin:/usr/bin:/usr/sbin"],
+		"args": ["freebsd-version"]
+	}
+	"mounts": [
+		{
+			"destination": "/dev",
+			"options": ["ruleset=4"],
+			"source": "devfs",
+			"type": "devfs"
+		},
+		{
+			"destination": "/dev/fd",
+			"source": "fdesc",
+			"type": "fdescfs"
+		}
+	],
+	"root": {
+		"path": "/path/to/container/root"
+	},
+	"freebsd": {
+		"jail": {
+			"host": "new",
+			"vnet": "new",
+			"enforceStatfs": 1,
+			"allow": {
+				"rawSockets": true,
+				"chflags": true,
+				"mount": [
+					"tmpfs"
```

Comment:
Given that we didn't specify a tmpfs in `mounts`, this allows the jail itself to create new tmpfs mounts?

Reply:
Correct.